### DISCUSSION TITLE
Turborepo - dev task - Are there hooks for apps to indicate to turbo it should shut down the dev instance?

### DISCUSSION THREAD

**Discussion by SReject** [vercel/turborepo#3873] [Help] [answered]
Is there any mechanism(s) where processes started by tubrorepo via `turbo run dev` can indicate to turborepo to shutdown turbo's entire dev instance?

For context, I'm working within an [electron](https://www.electronjs.org/)-based monorepo. When the main window is manually closed, I'd like to indicate to turbo the dev instance, as a whole, is to exit and have it shut down other dev-started instances.

**Comment by anthonyshew** [comment]
This would get covered by a "watch" feature. You can upvote this issue and follow #986  if you haven't already.

**Reply by SReject** [reply]
As I understand it, watch mode targets files. My question pretains to processes, initiated by turbo, being able to tell turbo to end all processes that were started.

**Reply by anthonyshew** [reply, marked as answer]
Ah, I see what you're describing now. An almost "reverse watch," where your task is going to kill `turbo`. Apologies for misunderstanding.

The short answer I'll give here is no, there is no way to do this today. If you feel strongly that you'd like this feature, please make us an issue for a feature request. 🙏
